### PR TITLE
Refactor- lint pacification

### PR DESCRIPTION
**Pull request by buddhamagnet**
Hope this helps, ran the code through ```golint``` and made the appropriate adjustments, some notes:

1. Couldn't pass the whole linter as some of the ```blackfriday``` interface functions also need attention.
2. Adjusted some methods returninh unexported types at the linter called that annoying :).
3. Adjusted all tests after the fact, all passing.

### CODE REVIEW

**Comment by mholt** [comment]
Thanks for doing this. It's tedious work so I really appreciate it.

I have mixed feelings about the linter, and I don't typically agree with all the notices. I'll make comments in-line soon, so look for those.

Also, we're about to merge the letsencrypt branch into master which is probably going to be a big headache, and this will only multiply the merge conflicts. So even though I like most of these changes, I'm afraid now is not quite the time to merge them into this branch. You will probably have some rebasing to do soon, just a heads-up.

File: config/parse/parsing.go [hunk 1]
```diff
@@ -301,9 +301,9 @@ func standardAddress(str string) (host, port string, err error) {
 }
 
 type (
-	// serverBlock associates tokens with a list of addresses
+	// ServerBlock associates tokens with a list of addresses
 	// and groups tokens by directive name.
-	serverBlock struct {
+	ServerBlock struct {
```
Comment (mholt) [inline config/parse/parsing.go]: Let's leave this type unexported.

File: config/setup/bindhost.go [hunk 2]
```diff
@@ -2,6 +2,7 @@ package setup
 
 import "github.com/mholt/caddy/middleware"
 
+// BindHost binds hosts to controllers.
```
Comment (mholt) [inline config/setup/bindhost.go]: Change to:

> BindHost sets the host to bind the listener to.

File: config/setup/root.go [hunk 3]
```diff
@@ -7,6 +7,7 @@ import (
 	"github.com/mholt/caddy/middleware"
 )
 
+// Root is the root path controller.
```
Comment (mholt) [inline config/setup/root.go]: Change to:
> Root sets up the root file path of the server.

File: config/setup/startupshutdown.go [hunk 4]
```diff
@@ -8,10 +8,12 @@ import (
 	"github.com/mholt/caddy/middleware"
 )
 
+// Startup starts up caddy.
```
Comment (mholt) [inline config/setup/startupshutdown.go]: Change to:
> Startup registers a startup callback to execute during server start.

File: config/setup/startupshutdown.go [hunk 5]
```diff
@@ -8,10 +8,12 @@ import (
...
 func Startup(c *Controller) (middleware.Middleware, error) {
 	return nil, registerCallback(c, &c.Startup)
 }
 
+// Shutdown shuts down caddy.
```
Comment (mholt) [inline config/setup/startupshutdown.go]: Change to:
> Shutdown registers a shutdown callback to execute during process exit.

File: config/setup/tls.go [hunk 6]
```diff
@@ -8,6 +8,7 @@ import (
 	"github.com/mholt/caddy/middleware"
 )
 
+// TLS supports TLS mode.
```
Comment (mholt) [inline config/setup/tls.go]: Change to:
> TLS sets up the TLS configuration (but does not activate Let's Encrypt; that is handled elsewhere).

File: middleware/fastcgi/fcgiclient_test.go [hunk 7]
```diff
@@ -194,7 +194,6 @@ func generateRandFile(size int) (p string, m string) {
 func DisabledTest(t *testing.T) {
 	// TODO: test chunked reader
 
-	t_ = t
```
Comment (mholt) [inline middleware/fastcgi/fcgiclient_test.go]: Without this line, how does the sendFcgi function get access to `t`?

File: middleware/recorder.go [hunk 8]
```diff
@@ -8,13 +8,13 @@ import (
...
 
-// responseRecorder is a type of ResponseWriter that captures
+// ResponseRecorder is a type of ResponseWriter that captures
 // the status code written to it and also the size of the body
 // written in the response. A status code does not have
 // to be written, however, in which case 200 must be assumed.
 // It is best to have the constructor initialize this type
 // with that default status code.
-type responseRecorder struct {
+type ResponseRecorder struct {
```
Comment (mholt) [inline middleware/recorder.go]: Let's leave this type unexported.

**Comment by buddhamagnet** [comment]
No worries @mholt I generally get involved first with pull requests like this, it's a good way to say hi :)

**Comment by mholt** [comment]
Indeed. :smile: I think this is looking good now. But I will let this stay open until letsencrypt is merged in and we're ready to make these changes.

**Comment by mholt** [comment]
@buddhamagnet I completed the merge. I suspect this will change at least some of your linting... sorry about that. Will welcome an updated PR at your leisure though.

File: config/config.go [hunk 9]
```diff
@@ -15,8 +15,11 @@ import (
 )
 
 const (
+	// DefaultHost determines the default host.
 	DefaultHost = "0.0.0.0"
+	// DefaultPort determines the defaulr port.
```
Comment (lenovouser) [inline config/config.go]: `defaulr` should be `default` :smiley:

**Comment by buddhamagnet** [comment]
@mholt not bother at all, I started this :)

I've rebased against master and force pushed, and passed all linter warnings now apart from these, which I think we can leave:

screen shot 2015-11-04 at 14 32 19

File: caddy/config.go [hunk 10]
```diff
@@ -14,7 +14,6 @@ import (
 	"github.com/mholt/caddy/server"
 )
 
-const (
```
Comment (mholt) [inline caddy/config.go]: Oops

You might need that

**Comment by mholt** [comment]
Make sure you pull the latest; looks like a test is still failing.

**Comment by buddhamagnet** [comment]
@mholt that should be it buddy.

**Comment by mholt** [comment]
Great! Thank you.

**Pull request merged**